presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -630,6 +672,39 @@ protected boolean internalStartNext()
         }
     }
 
+    protected void enforceTimeouts()
+    {
+        checkState(Thread.holdsLock(root), "Must hold lock to find next query");
+        synchronized (root) {
+            for (InternalResourceGroup group : subGroups.values()) {
+                group.enforceTimeouts();
+            }
+            long runningTimeoutMillis = runningTimeout.toMillis();
+            long queuedTimeoutMillis = queuedTimeout.toMillis();
+            //log.info("Group %s: %d, %d", id, queuedTimeoutMillis, runningTimeoutMillis);
+            for (QueryExecution query : ImmutableList.copyOf(activeQueries)) {
+                Duration elapsedTime = query.getQueryInfo().getQueryStats().getElapsedTime();
+                if (elapsedTime == null) {
```

Comment:
Under what conditions does this happen? Should you really be checking stats instead of state?